integration-tests/src/test/java/org/apache/fineract/integrationtests/accounting/AccountingRuleIntegrationTest.java

```diff
@@ -0,0 +1,79 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.fineract.integrationtests.accounting;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+import io.restassured.builder.RequestSpecBuilder;
+import io.restassured.builder.ResponseSpecBuilder;
+import io.restassured.http.ContentType;
+import io.restassured.specification.RequestSpecification;
+import io.restassured.specification.ResponseSpecification;
+import java.util.ArrayList;
+import org.apache.fineract.client.models.GetAccountRulesResponse;
+import org.apache.fineract.client.models.GetOfficesResponse;
+import org.apache.fineract.client.models.PostAccountingRulesResponse;
+import org.apache.fineract.integrationtests.common.OfficeHelper;
+import org.apache.fineract.integrationtests.common.Utils;
+import org.apache.fineract.integrationtests.common.accounting.Account;
+import org.apache.fineract.integrationtests.common.accounting.AccountHelper;
+import org.apache.fineract.integrationtests.common.accounting.AccountRuleHelper;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class AccountingRuleIntegrationTest {
+
+    private ResponseSpecification responseSpec;
+    private RequestSpecification requestSpec;
+
+    private AccountHelper accountHelper;
+    private AccountRuleHelper accountRuleHelper;
+
+    @BeforeEach
+    public void setup() {
+        Utils.initializeRESTAssured();
+
+        requestSpec = new RequestSpecBuilder().setContentType(ContentType.JSON).build();
+        requestSpec.header("Authorization", "Basic " + Utils.loginIntoServerAndGetBase64EncodedAuthenticationKey());
+        responseSpec = new ResponseSpecBuilder().expectStatusCode(200).build();
+
+        accountRuleHelper = new AccountRuleHelper(requestSpec, responseSpec);
+        accountHelper = new AccountHelper(requestSpec, responseSpec);
+    }
+
+    @Test
+    public void testAccountingRuleCreation() {
+        // given
+        final Account accountToCredit = accountHelper.createIncomeAccount();
+        final Account accountToDebit = accountHelper.createExpenseAccount();
+        final GetOfficesResponse headOffice = OfficeHelper.getHeadOffice(requestSpec, responseSpec);
+
+        // when
+        final PostAccountingRulesResponse accountingRule = accountRuleHelper.createAccountRule(headOffice.getId(), accountToCredit,
+                accountToDebit);
+        assertNotNull(accountingRule);
+        assertNotNull(accountingRule.getResourceId());
+
+        // then
+        final ArrayList<GetAccountRulesResponse> accountingRules = accountRuleHelper.getAccountingRules();
+        assertNotNull(accountingRules);
+        assertEquals(1, accountingRules.size());
```

Comment:
This is not recommended way of checking existence of a created entry. You cannot be sure whether it was not there already earlier, also you cannot be sure whether any other integration test will not created an account rules as well and if that one is executed earlier, this one will fail.

My recommendation is to check the actual id and some of the most important part of the account rules details.

Reply:
In this case, there were not any Integration Test for Accounting rules, so It is brand new, to cover the test for the Get Accounting Rules API that was failing with the SQL using 0 and 1 instead of boolean values as It was fixed in the PR https://github.com/apache/fineract/pull/2335

<img width="1115" alt="Screen Shot 2022-06-28 at 16 32 09" src="https://user-images.githubusercontent.com/44206706/176302006-582c37ba-85d4-4b2c-8542-19de777643ba.png">

Reply:
I understand there is no any integration tests for accounting rules...yet. 
1. But what if i created one tomorrow and as the execution order of the tests are not specified it could happen that the new test is executed earlier and if that integration test is creating an accounting rule, your test will fail.
2. You are not testing whether the accounting rule were created properly as there are missing checks for:
- officeId
- accountCreditId
- accountDebitId
- name
- description

My recommendation is to avoid checks like:
- assertEquals(1, accountingRules.size());

And preferrably find the created accounting rule entry and check its details one by one

Reply:
Done!